### DISCUSSION TITLE
Do I have to publish CDK construct library?

### DISCUSSION THREAD

**Discussion by shawnwang-avl** [projen/projen#2061] [Q&A]
Hi,

I'm trying to setup a simple CDK construct library and CDK app. The only examples I've been able to find is to publish library release to any package manager, and depend on the new version in CDK app. Can I skip the package manager? I'd just like the CDK app to depend on a major version (e.g., `~1`) of library, and it should just use the latest on `main`. Is this possible with projen?

**Comment by pflorek** [comment]
Hey @shawnwang-avl ,

the project awscdk-construct project type's main purpose is to have a quickstart to manage a project for reusable constructs. 

Using a package manager to share the library across projects is convenient. You don't have to use that feature, you may use a private registry and alternatives like git submodule or use npm/yarn to depend on a different location. 

For the awscdk-construct the configured cdkVersion will be the minimal required version in your downstream project.

If you only need an cdk app, then use one of the `awscdk-app` starters. Yes, I assume it's also working by setting the cdkVersion to latest, since it should be a valid version alias on npm.

**Comment by shawnwang-avl** [comment]
Thanks @pflorek . We are trying to **avoid** private registry at my company. How would I go about omitting all of the publishing and configuring git submodule? Do I use `deps: []` ([link](https://projen.io/awscdk-construct.html#dependencies))?

**Comment by mrgrain** [comment]
Docs are a bit all over the place for this, but see:
https://projen.io/publisher.html#dry-run
https://projen.io/releases.html#manual-releases


Private registry:
https://projen.io/publisher.html#publishing-to-aws-codeartifact

**Comment by shawnwang-avl** [comment]
sorry typo avoid private registry.
We just want to use git submodule.

**Reply by mrgrain** [reply]
I'm not super familiar with `git submodule`. As far as I understand them there's no publishing involved. You should be able to simply disable all publishing and use `git submodule` as you usually would with npm etc.

There is no explicit feature or support for `git submodules` in projen at the moment.
